gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowConfigsResource.java

```diff
@@ -124,12 +124,38 @@ public UpdateResponse update(ComplexResourceKey<FlowId, EmptyRecord> key, FlowCo
    */
   @Override
   public UpdateResponse delete(ComplexResourceKey<FlowId, EmptyRecord> key) {
+    checkRequester(get(key), this.requesterService.findRequesters(this));
     String flowGroup = key.getKey().getFlowGroup();
     String flowName = key.getKey().getFlowName();
     FlowId flowId = new FlowId().setFlowGroup(flowGroup).setFlowName(flowName);
     return this.flowConfigsResourceHandler.deleteFlowConfig(flowId, getHeaders());
   }
 
+  /**
+   * Check that all {@link ServiceRequester}s in this request are contained within the original service requester list
+   * when the flow was submitted and throw a {@link FlowConfigLoggedException} if they are not.
+   *
+   * @param originalFlowConfig original flow config to find original requester
+   * @param requesterList list of requesters for this request
+   */
+  public static void checkRequester(FlowConfig originalFlowConfig, List<ServiceRequester> requesterList) {
```

Comment:
I think this method should return a boolean.
Or at least javadoc needs to have information on what it throws and what that means.

Reply:
I think I still prefer throwing exception in the method to avoid having to copy the throwing exception code to five places. I updated the javadoc to make the exceptions thrown more clear.